federated-xgboost/README.md

```diff
@@ -1,12 +1,33 @@
 # federated-xgboost
 Federated learning allows multiple parties to collaboratively learn a shared model while keeping each party's data at its respective site. It allows for collaborative learning with lower latencies without a central data storage, thereby improving the privacy of individual parties' data.
 
-In the federated setting, a central party has a basic model that is initially broadcast to all parties. Each party locally trains the model with its own data, then sends a summary of the updates to the model back to the central party. In the decision tree case, parties would be sending the best local feature splits back to the central party. The central party then aggregates all updates, updates its own model with the aggregated update, and broadcasts the newly updated model to all parties. This process is then repeated over and over.
-
-![federated diagram](./images/federated-xgboost-diagram.png)
+In the federated setting, a central party has a basic model that is initially broadcast to all parties. Each party locally trains the model with its own data, then sends a summary of the updates to the model back to the central party. In the decision tree case, parties would be sending the best local feature splits back to the central party. The central party then aggregates all updates, updates its own model with the aggregated update, and broadcasts the newly updated model to all parties. This process is then repeated over and over. See the diagrams below for a visualization of this workflow.
 
 This project extends the existing XGBoost gradient boosting machine learning framework to enable training models in the federated setting. This work is being actively contributed to and is still under development.
 
+[QUICKSTART](#quickstart)
+
+### _High Level_
+
+![federated diagram](./images/federated-xgboost-diagram.png)
+
+### _Finding a Split_
+ 
+ ![diagram 1](./images/detail_1.png)
+
+ ![diagram 2](./images/detail_2.png)
+
+ ![diagram 3](./images/detail_3.png)
+ 
+ ![diagram 4](./images/detail_4.png)
```

Comment:
Can you try left aligning the steps so that they're all in the same place? Not sure if that'll look better but the steps look a little strange right now. 

I also wonder whether the README is the best place to put this. Maybe we can have a link in the README to a different file that shows this diagram. What are your thoughts?

Reply:
Yeah, I think thats better than my awkward jump to Quickstart link thing